### PR TITLE
modernize plugin

### PR DESCRIPTION
**Pull request by mawinter69**
enable cd, update to current Jenkins

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+_extends: .github
```
Comment (jglick) [inline .github/release-drafter.yml:1]: This file is not actually needed. (see migration docs)

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+buildPlugin(
+  useContainerAgent: true,
+  configurations: [
+    [platform: 'linux', jdk: 11]
```
Comment (jglick) [inline Jenkinsfile:4]: Deliberately skipping Windows, and Java 17/21?

File: pom.xml [hunk 3]
```diff
@@ -5,25 +5,30 @@
...
     <license>
       <name>MIT License</name>
-      <url>http://opensource.org/licenses/MIT</url>
+      <url>https://opensource.org/licenses/MIT</url>
     </license>
   </licenses>
 
+  <properties>
+    <changelist>999999-SNAPSHOT</changelist>
+    <jenkins.version>2.423</jenkins.version>
```
Comment (jglick) [inline pom.xml:29]: Why the weekly dep?

File: pom.xml [hunk 4]
```diff
@@ -35,28 +40,39 @@
...
-      <url>http://repo.jenkins-ci.org/public/</url>
+      <url>https://repo.jenkins-ci.org/public/</url>
     </pluginRepository>
   </pluginRepositories>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>io.jenkins.tools.bom</groupId>
+        <artifactId>bom-weekly</artifactId>
```
Comment (jglick) [inline pom.xml:64]: Should not normally be used as a dep.

**Review by jglick — COMMENTED** [review]
Not sure why I am asked to review this. Do you plan to adopt the plugin?

**Comment by mawinter69** [comment]
> Not sure why I am asked to review this. Do you plan to adopt the plugin?
You're one of the 2 maintainers and the other seems to no longer do anything here.
But I would volunteer to adopt it.

File: Jenkinsfile (hunk 2 above)
Comment (mawinter69) [reply Jenkinsfile:4]: for now yes, there is no code where I see the need to run on windows or use a newer jdk (maybe one can enable jdk21 though)

File: pom.xml (hunk 3 above)
Comment (mawinter69) [reply pom.xml:29]: mostly because of #5

File: pom.xml (hunk 4 above)
Comment (mawinter69) [reply pom.xml:64]: it's only needed for the tests, I plan to go back to an LTS once the change in core https://github.com/jenkinsci/jenkins/pull/8496 is available in an LTS release

Comment (jglick) [reply pom.xml:64]: OK then this should be in draft, or you should revert the weekly changes here so it is purely about updating plugin metadata to a recommended baseline and then have a separate draft PR taking advantage of that core change.

File: pom.xml [hunk 5]
```diff
@@ -5,25 +5,30 @@
...
     <license>
       <name>MIT License</name>
-      <url>http://opensource.org/licenses/MIT</url>
+      <url>https://opensource.org/licenses/MIT</url>
     </license>
   </licenses>
 
+  <properties>
+    <changelist>999999-SNAPSHOT</changelist>
+    <jenkins.version>2.414.1</jenkins.version>
```
Comment (jglick) [inline pom.xml:29]: https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/ FWIW

**Review by jglick — COMMENTED** [review]
Best to ask to adopt this plugin in RPU. Looks like I made some change eight years ago but I am definitely not thinking of myself as a maintainer! :grin:

**Pull request merged**